Options.py

```diff
@@ -708,6 +707,29 @@ def from_text(cls, text: str) -> Range:
             return cls(cls.special_range_names[text])
         return super().from_text(text)
 
+    @classmethod
+    def weighted_range(cls, text) -> Range:
+        if text == "random-low":
+            return cls(cls.triangular(cls.range_start, cls.range_end, cls.range_start))
+        elif text == "random-high":
+            return cls(cls.triangular(cls.range_start, cls.range_end, cls.range_end))
+        elif text == "random-middle":
+            return cls(cls.triangular(cls.range_start, cls.range_end))
+        elif text.startswith("random-range-"):
+            return cls.custom_range(text)
+        elif text == "random":
+            return cls(random.randint(cls.range_start, cls.range_end))
+        else:
+            raise Exception(f"random text \"{text}\" did not resolve to a recognized pattern. "
+                            f"Acceptable values are: random, random-high, random-middle, random-low, "
+                            f"random-range-low-<min>-<max>, random-range-middle-<min>-<max>, "
+                            f"random-range-high-<min>-<max>, or random-range-<min>-<max>.")
+
+
+class SpecialRange(NamedRange):
```

Comment:
a unit test or at least an assert should be added to this class that it's being used correctly and catches the existing world bugs.

Reply:
I don't necessarily agree with force-failing a bunch of worlds that are probably behaving fine, just in case. I'd rather tell them about the possible change once it's in, and if they do fix it, an assert could come later